corehq/apps/app_manager/suite_xml/post_process/remote_requests.py

```diff
@@ -254,15 +265,74 @@ def build_query_prompts(self):
 
     def build_stack(self):
         stack = Stack()
-        frame = PushFrame()
+        rewind_if = None
+        if module_uses_smart_links(self.module):
+            user_domain_xpath = session_var(COMMCARE_PROJECT, path="user/data")
+            # For case in same domain, do a regular case claim rewind
+            rewind_if = self._get_case_domain_xpath().eq(user_domain_xpath)
+            # For case in another domain, jump to that other domain
+            frame = PushFrame(if_clause=self._get_case_domain_xpath().neq(user_domain_xpath))
```

Comment:
nit: perhaps this is personal preference and I know that these are logically the same but I think this is clearer that one is the inverse of the other:
```suggestion
            frame = PushFrame(if_clause=XPath.not_(rewind_if))
```

Reply:
thanks, that's much more readable